Lib/test/test_random.py

```diff
@@ -1104,13 +1104,13 @@ def test_binomialvariate(self):
         # exceedingly unlikely to ever fail for correct code.
 
         # BG code path
-        # Expected dist: [31641, 42188, 21094, 4688, 391]
+        # Expected dist: [42139, 42317, 13979, 1565]
         c = Counter(B(4, 0.25) for i in range(100_000))
-        self.assertTrue(29_641 <= c[0] <= 33_641, c)
-        self.assertTrue(40_188 <= c[1] <= 44_188)
-        self.assertTrue(19_094 <= c[2] <= 23_094)
-        self.assertTrue(2_688  <= c[3] <= 6_688)
-        self.assertEqual(set(c), {0, 1, 2, 3, 4})
+        self.assertTrue(40_139 <= c[0] <= 44_139, c)
+        self.assertTrue(40_317 <= c[1] <= 44_317)
+        self.assertTrue(11_979 <= c[2] <= 15_979)
+        self.assertTrue(0      <= c[3] <= 3_565)
+        self.assertEqual(set(c), {0, 1, 2, 3})
```

Comment:
This is definitely wrong. `B(n, p)` should has value `{0, 1, 2, ..., n}`

Reply:
Binomial distrubution table for `B(4,p)`

p | .01 | .05 | .10 | .15 | .20 | .25 | .30 | .35 | .40 | .45 | .50 | .55 | .60 | .65 | .70 | .75 | .80 | .85 | .90 | .95
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
r | 0 | .961 | .815 | .656 | .522 | .410 | .316 | .240 | .179 | .130 | .092 | .062 | .041 | .026 | .015 | .008 | .004 | .002 | .001 | .000 | .000
  | 1 | .039 | .171 | .292 | .368 | .410 | .422 | .412 | .384 | .346 | .300 | .250 | .200 | .154 | .112 | .076 | .047 | .026 | .011 | .004 | .000
  | 2 | .001 | .014 | .049 | .098 | .154 | .211 | .265 | .311 | .346 | .368 | .375 | .368 | .346 | .311 | .265 | .211 | .154 | .098 | .049 | .014
  | 3 | .000 | .000 | .004 | .011 | .026 | .047 | .076 | .112 | .154 | .200 | .250 | .300 | .346 | .384 | .412 | .422 | .410 | .368 | .292 | .171
  | 4 | .000 | .000 | .000 | .001 | .002 | .004 | .008 | .015 | .026 | .041 | .062 | .092 | .130 | .179 | .240 | .316 | .410 | .522 | .656 | .815

Taken from https://www.thoughtco.com/binomial-table-n-2-through-6-3126258